lib/scope.js

```diff
@@ -1805,34 +1805,35 @@ module.exports = {
     /* Try to upload one or more files to a file-type input field.
     *    Does not submit, so no danger of timeout with server reload.
     */
-    
-    // Tests could be in one of two places. Figure out which directory they're in
-    // Makes an absolute path from the directory in which `npm run` was used plus our internal folders
-    let dir_source = `${ process.cwd() }/docassemble/${ session_vars.get_repo_url().match(/docassemble-(.*)/)[1] }/data/sources`;
-    let dir_features = `${ process.cwd() }/tests/features`;
-    let dir = null;
 
-    // Make sure one of the valid directories exist. This only allows
-    // one directory to be used per Step or table row. We don't anticipate
-    // future people having a `tests/features` folder, let alone both.
+    // Tests should be one place: docassemble/*/data/sources, where * is the only directory in `docassemble`.
+    // Makes an absolute path from the directory in which `npm run` was used plus our internal folders
+    let docassemble_pkg_path = `${ process.cwd() }/docassemble`;
+    let pkg_path = null;
     try {
-      // Try the most common directory first
-      fs.accessSync( dir_source );
-      dir = dir_source;
-      // If they have both directories, let them know which directory will be used.
-      try {
-        fs.accessSync( dir_features );
-        await scope.addToReport( scope, { type: `warning`, value: `You have both a docassemble package 'sources' folder and a 'test/features' folder. These tests will use the package's 'sources' folder.` });
-      } catch ( error_features ) { `all's well, do nothing`; }
-
-    } catch ( error_source ) {
-      try {
-        fs.accessSync( dir_features );
-        dir = dir_features;
-      } catch ( error_features ) {
-        await scope.addToReport( scope, { type: `warning`, value: `There are two places where you can store files to upload: "${ dir_source }" and "${ dir_features }". These tests could not find either of those.` });
-        throw error_source;  // Most people will want to know the sources folder path;
+      fs.accessSync(docassemble_pkg_path);
+      // https://stackoverflow.com/a/24594123/11416267
+      let dirs = fs.readdirSync(docassemble_pkg_path, {withFileTypes: true}).filter(dirent => dirent.isDirectory()).map(dirent => dirent.name);
+      if (dirs.length < 1) {
+        let msg = `You can store files to upload in 'docassemble/*/data/source'. These tests could not find a directory like that.`;
+        await scope.addToReport( scope, { type: `warning`, value: msg});
+        throw Error(msg);  // Most people will want to know the sources folder path;
+      }
+      if (dirs.length != 1) {
+        await scope.addToReport( scope, { type: `warning`, value: `You have multiple directories in 'docassemble' folder (${ dirs }). These tests will use the first directory (${dirs[0]}).` });
```

Comment:
Nitpick: Instead of multiple directories, would it be appropriate to say multiple package folders or multiple project folders? I think new developers would be confused by this. I also think we might simplify language a bit by using 'folders' instead of 'directories' in various places. I've probably used 'directories' myself in other places and never noticed it, but I think 'folders' is more clear.

Reply:
Changed to just "folders", don't think there's much benefit in trying to be more specific with "subpackage folder" or anything like that.